Sources/Purchasing/ConfigurationInCustomEntitlementsComputation.swift

```diff
@@ -0,0 +1,229 @@
+
+import Foundation
+
+#if ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
+
+// swiftlint:disable type_name
+/**
+ * ``ConfigurationInCustomEntitlementsComputation`` can be used when configuring the ``Purchases`` instance.
+ * It is not required to be used, but highly recommended. This class follows a builder pattern.
+ *
+ * To configure your `Purchases` instance using this object, follow these steps.
+ *
+ * **Steps:**
+ * 1. Call ``ConfigurationInCustomEntitlementsComputation/builder(withAPIKey:)``
+ *  To obtain a ``ConfigurationInCustomEntitlementsComputation/Builder`` object.
+ * 2. Set this builder's properties using the "`with(`" functions.
+ * 3. Call ``ConfigurationInCustomEntitlementsComputation/Builder/build()`` to obtain the
+ *  `ConfigurationInCustomEntitlementsComputation` object.
+ * 4. Pass the `ConfigurationInCustomEntitlementsComputation` object into
+ *  ``Purchases/configureInCustomEntitlementsComputationMode(with:)``.
+ *
+ * ```swift
+ * let configuration = ConfigurationInCustomEntitlementsComputation.Builder(withAPIKey: "MyKey",
+ *                                                                          appUserID: "SomeAppUserID).build()
+ *  Purchases.configure(with: configuration)
+ * ```
+ */
+public final class ConfigurationInCustomEntitlementsComputation: NSObject {
+
+    static let storeKitRequestTimeoutDefault: TimeInterval = 30
+    static let networkTimeoutDefault: TimeInterval = 60
+
+    let apiKey: String
+    let appUserID: String?
+    internal let observerMode: Bool
+    internal let userDefaults: UserDefaults?
+    let storeKitVersion: StoreKitVersion
+    internal let dangerousSettings: DangerousSettings?
+    internal let networkTimeout: TimeInterval
+    internal let storeKit1Timeout: TimeInterval
+    internal let platformInfo: Purchases.PlatformInfo?
+    internal let responseVerificationMode: Signing.ResponseVerificationMode
+    let showStoreMessagesAutomatically: Bool
+    internal let diagnosticsEnabled: Bool
+
+    private init(with builder: Builder) {
+        Self.verify(apiKey: builder.apiKey)
+
+        self.apiKey = builder.apiKey
+        self.appUserID = builder.appUserID
+        self.observerMode = builder.observerMode
+        self.userDefaults = builder.userDefaults
+        self.storeKitVersion = builder.storeKitVersion
+        self.dangerousSettings = builder.dangerousSettings
+        self.storeKit1Timeout = builder.storeKit1Timeout
+        self.networkTimeout = builder.networkTimeout
+        self.platformInfo = builder.platformInfo
+        self.responseVerificationMode = builder.responseVerificationMode
+        self.showStoreMessagesAutomatically = builder.showStoreMessagesAutomatically
+        self.diagnosticsEnabled = builder.diagnosticsEnabled
+    }
+
+    /// Factory method for the ``Configuration/Builder`` object that is required to create a `Configuration`
+    @objc public static func builder(withAPIKey apiKey: String, appUserID: String) -> Builder {
```

Comment:
Related to my previous comment. Is `@objc` needed here?

Reply:
Does not apply anymore after removing this class